```
Error running mint automation
+ mkdir -p 16201-d6a8144/gopath/src/github.com/minio
+ git -C 16201-d6a8144/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 16201-d6a8144/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 16201-d6a8144/gopath/src/github.com/minio/minio fetch --quiet minio pull/16201/head:pr16201
+ git -C 16201-d6a8144/gopath/src/github.com/minio/minio checkout --quiet pr16201
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/16201-d6a8144/gopath
+ make -C 16201-d6a8144/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/minio/madmin-go/v2 v2.0.1
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/bits-and-blooms/bloom/v3 v3.3.1
go: downloading github.com/minio/console v0.22.2
go: downloading github.com/minio/minio-go/v7 v7.0.44
go: downloading github.com/dchest/siphash v1.2.3
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading go.uber.org/zap v1.23.0
go: downloading github.com/djherbis/atime v1.1.0
go: downloading go.etcd.io/etcd/client/v3 v3.5.5
go: downloading github.com/minio/dperf v0.4.2
go: downloading golang.org/x/time v0.0.0-20220722155302-e5dcc9cfc0b9
go: downloading cloud.google.com/go/storage v1.26.0
go: downloading github.com/coreos/go-oidc v2.2.1+incompatible
go: downloading github.com/minio/kes v0.22.0
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/klauspost/compress v1.15.12
go: downloading github.com/shirou/gopsutil/v3 v3.22.9
go: downloading github.com/google/uuid v1.3.0
go: downloading go.uber.org/atomic v1.10.0
go: downloading github.com/prometheus/client_model v0.3.0
go: downloading github.com/prometheus/procfs v0.8.0
go: downloading golang.org/x/crypto v0.3.0
go: downloading github.com/prometheus/client_golang v1.13.0
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/inconshreveable/mousetrap v1.0.1
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/minio/zipindex v0.3.0
go: downloading github.com/zeebo/xxh3 v1.0.2
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/pierrec/lz4 v2.6.1+incompatible
go: downloading github.com/golang-jwt/jwt/v4 v4.4.2
go: downloading github.com/minio/xxml v0.0.3
go: downloading github.com/philhofer/fwd v1.1.2-0.20210722190033-5c56ac6d0bb9
go: downloading github.com/minio/cli v1.24.0
go: downloading github.com/lithammer/shortuuid/v4 v4.0.0
go: downloading github.com/klauspost/reedsolomon v1.11.3
go: downloading github.com/prometheus/common v0.37.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/miekg/dns v1.1.50
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/tinylib/msgp v1.1.7-0.20220719154719-f3635b96e483
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading go.etcd.io/etcd/api/v3 v3.5.5
go: downloading github.com/go-openapi/loads v0.21.2
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/minio/selfupdate v0.5.0
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/klauspost/filepathx v1.1.1
go: downloading golang.org/x/oauth2 v0.2.0
go: downloading golang.org/x/sys v0.2.0
go: downloading google.golang.org/api v0.98.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/coredns/coredns v1.9.4
go: downloading github.com/felixge/fgprof v0.9.3
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/rs/cors v1.8.2
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/Azure/azure-storage-blob-go v0.15.0
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading github.com/Shopify/sarama v1.36.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.4.1
go: downloading github.com/elastic/go-elasticsearch/v7 v7.17.7
go: downloading github.com/go-sql-driver/mysql v1.6.0
go: downloading github.com/gomodule/redigo v1.8.9
go: downloading github.com/lib/pq v1.10.7
go: downloading github.com/nats-io/nats.go v1.17.0
go: downloading github.com/nats-io/stan.go v0.10.3
go: downloading github.com/nsqio/go-nsq v1.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/rabbitmq/amqp091-go v1.5.0
go: downloading github.com/xdg/scram v1.0.5
go: downloading github.com/go-ldap/ldap/v3 v3.4.4
go: downloading github.com/minio/simdjson-go v0.4.2
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.16
go: downloading golang.org/x/net v0.2.0
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/mattn/go-runewidth v0.0.14
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/fraugster/parquet-go v0.12.0
go: downloading github.com/pquerna/cachecontrol v0.1.0
go: downloading gopkg.in/square/go-jose.v2 v2.6.0
go: downloading github.com/alecthomas/participle v0.7.1
go: downloading github.com/klauspost/cpuid/v2 v2.2.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading golang.org/x/sync v0.1.0
go: downloading github.com/xdg/stringprep v1.0.3
go: downloading github.com/rivo/uniseg v0.4.2
go: downloading github.com/bits-and-blooms/bitset v1.3.3
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20220621081337-cb9428e4ac1e
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.4
go: downloading github.com/golang/snappy v0.0.4
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.5
go: downloading google.golang.org/grpc v1.50.1
go: downloading google.golang.org/genproto v0.0.0-20221018160656-63c7b68cfc55
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/google/pprof v0.0.0-20220829040838-70bd9ae97f40
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/Azure/azure-pipeline-go v0.2.3
go: downloading golang.org/x/text v0.4.0
go: downloading go.uber.org/multierr v1.8.0
go: downloading github.com/gorilla/websocket v1.5.0
go: downloading google.golang.org/protobuf v1.28.1
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/rs/xid v1.4.0
go: downloading github.com/go-openapi/analysis v0.21.4
go: downloading github.com/go-openapi/spec v0.20.7
go: downloading github.com/go-openapi/swag v0.22.3
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading github.com/tklauser/go-sysconf v0.3.11
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading aead.dev/mem v0.2.0
go: downloading cloud.google.com/go v0.104.0
go: downloading github.com/coreos/go-systemd/v22 v22.4.0
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading github.com/tklauser/numcpus v0.6.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/jcmturner/gofork v1.7.6
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.3
go: downloading github.com/eapache/go-resiliency v1.3.0
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pierrec/lz4/v4 v4.1.16
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/apache/thrift v0.16.0
go: downloading github.com/hashicorp/go-uuid v1.0.3
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/go-openapi/strfmt v0.21.3
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/go-openapi/jsonreference v0.20.0
go: downloading github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
go: downloading go.mongodb.org/mongo-driver v1.10.3
go: downloading github.com/go-openapi/errors v0.20.3
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/oklog/ulid v1.3.1
go: downloading cloud.google.com/go/compute v1.10.0
go: downloading cloud.google.com/go/iam v0.4.0
go: downloading go.opencensus.io v0.23.0
go: downloading github.com/googleapis/gax-go/v2 v2.5.1
go: downloading golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/go-openapi/runtime v0.24.2
go: downloading github.com/go-openapi/validate v0.22.0
go: downloading github.com/minio/mc v0.0.0-20221201184114-854b4f123f03
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/unrolled/secure v1.13.0
go: downloading github.com/tidwall/gjson v1.14.3
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/tidwall/pretty v1.2.1
go: downloading github.com/docker/go-units v0.5.0
go: downloading github.com/jedib0t/go-pretty/v6 v6.3.8
go: downloading github.com/gdamore/tcell/v2 v2.5.3
go: downloading github.com/charmbracelet/bubbles v0.14.0
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading github.com/rivo/tview v0.0.0-20220916081518-2e69b7385a37
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/minio/filepath v1.0.0
go: downloading github.com/posener/complete v1.2.3
go: downloading github.com/pkg/xattr v0.4.9
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/charmbracelet/bubbletea v0.22.1
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/charmbracelet/lipgloss v0.6.0
go: downloading github.com/navidys/tvxwidgets v0.1.1
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading golang.org/x/term v0.2.0
go: downloading github.com/minio/colorjson v1.0.4
go: downloading github.com/prometheus/prom2json v1.3.2
go: downloading github.com/muesli/termenv v0.13.0
go: downloading github.com/muesli/cancelreader v0.2.2
go: downloading github.com/mattn/go-localereader v0.0.1
go: downloading github.com/containerd/console v1.0.3
go: downloading github.com/muesli/ansi v0.0.0-20211031195517-c9f0611b6c70
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/aymanbagabas/go-osc52 v1.2.1
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.1.0
go: downloading github.com/google/go-cmp v0.5.9
cmd/acl-handlers.go:28:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/restapi/user_session.go:30:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/pkg/certs/certs.go:33:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/pkg/logger/console.go:28:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/pkg/utils/parity.go:24:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/pkg/auth/token/config.go:23:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/admin-user-svcacct-add.go:32:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
internal/config/identity/ldap/config.go:28:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/subnet-utils.go:41:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/client-s3.go:51:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.22.2/restapi/client.go:30:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/config-fix.go:29:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
cmd/perf-tests.go:34:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
cmd/server-rlimit.go:26:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/main.go:43:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/find.go:38:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20221201184114-854b4f123f03/cmd/main.go:44:2: github.com/minio/pkg@v1.5.6: replacement directory ../minio-pkg does not exist
make: *** [build] Error 1

```

--- a/docs/sts/ldap.md
+++ b/docs/sts/ldap.md
@@ -34,7 +34,8 @@ KEY:
 identity_ldap  enable LDAP SSO support
 
 ARGS:
-MINIO_IDENTITY_LDAP_SERVER_ADDR*             (address)   AD/LDAP server address e.g. "myldapserver.com:636"
+MINIO_IDENTITY_LDAP_SERVER_ADDR*             (address)   AD/LDAP server address e.g. "myldap.com" or "myldapserver.com:1686"
+MINIO_IDENTITY_LDAP_SRV_RECORD_NAME          (string)    DNS SRV record name for LDAP service, if given, must be one of ldap, ldaps or on
 MINIO_IDENTITY_LDAP_LOOKUP_BIND_DN*          (string)    DN for LDAP read-only service account used to perform DN and group lookups
 MINIO_IDENTITY_LDAP_LOOKUP_BIND_PASSWORD     (string)    Password for LDAP read-only service account used to perform DN and group lookups
 MINIO_IDENTITY_LDAP_USER_DN_SEARCH_BASE_DN*  (list)      ";" separated list of user search base DNs e.g. "dc=myldapserver,dc=com"
@@ -52,17 +53,32 @@ MINIO_IDENTITY_LDAP_COMMENT                  (sentence)  optionally add a commen
 The variables relevant to configuring connectivity to the LDAP service are:
 
 ```
-MINIO_IDENTITY_LDAP_SERVER_ADDR*            (address)   AD/LDAP server address e.g. "myldapserver.com:636"
+MINIO_IDENTITY_LDAP_SERVER_ADDR*             (address)   AD/LDAP server address e.g. "myldap.com" or "myldapserver.com:1686"
+MINIO_IDENTITY_LDAP_SRV_RECORD_NAME          (string)    DNS SRV record name for LDAP service, if given, must be one of ldap, ldaps or on
 MINIO_IDENTITY_LDAP_TLS_SKIP_VERIFY         (on|off)    trust server TLS without verification, defaults to "off" (verify)
 MINIO_IDENTITY_LDAP_SERVER_INSECURE         (on|off)    allow plain text connection to AD/LDAP server, defaults to "off"
 MINIO_IDENTITY_LDAP_SERVER_STARTTLS         (on|off)    use StartTLS connection to AD/LDAP server, defaults to "off"
 ```
 
-The server address variable is _required_. TLS is assumed to be on by default.
+The server address variable is _required_. TLS is assumed to be on by default. The port in the server address is optional and defaults to 636 if not provided.
 
 **MinIO sends LDAP credentials to the LDAP server for validation. So we _strongly recommend_ to use MinIO with AD/LDAP server over TLS or StartTLS _only_. Using plain-text connection between MinIO and LDAP server means _credentials can be compromised_ by anyone listening to network traffic.**
 
 If a self-signed certificate is being used, the certificate can be added to MinIO's certificates directory, so it can be trusted by the server.
+
+#### DNS SRV Records
+
+Many Active Directory and other LDAP services are setup with [DNS SRV Records](https://ldap.com/dns-srv-records-for-ldap/) for high-availability of the directory service. To use this to find LDAP servers to connect to, an LDAP client makes a DNS SRV record request to the DNS service on a domain that looks like `_service._proto.example.com`. For LDAP the `proto` value is always `tcp`, and `service` is usually `ldap` or `ldaps`.
+
+To enable MinIO to use the SRV records, specify the `srv_record_name` config parameter (or equivalently the `MINIO_IDENTITY_LDAP_SRV_RECORD_NAME` environment variable). This parameter can be set to `ldap` or `ldaps` and MinIO will substitute it into the `service` value. For example, when `server_addr=myldapserver.com` and `srv_record_name=ldap`, MinIO will lookup the SRV record for `_ldap._tcp.myldapserver.com` and pick an appropriate target for LDAP requests.
+
+If the DNS SRV record is at an entirely different place, say `_ldapsrv._tcpish.myldapserver.com`, then set `srv_record_name` to the special value `on` and set `server_addr=_ldapsrv._tcpish.myldapserver.com`.
+
+When using this feature, do not specify a port in the `server_addr` as the port is picked up automatically from the SRV record.
+
+With the default (empty) value for `srv_record_name`, MinIO **will not** perform any SRV record request.
+
+The value of `srv_record_name` does not affect any TLS settings - they must be configured with their own parameters.
 
 ### Lookup-Bind
 

--- a/go.mod
+++ b/go.mod
@@ -51,7 +51,7 @@ require (
 	github.com/minio/kes v0.22.0
 	github.com/minio/madmin-go/v2 v2.0.1
 	github.com/minio/minio-go/v7 v7.0.44
-	github.com/minio/pkg v1.5.6
+	github.com/minio/pkg v1.5.8
 	github.com/minio/selfupdate v0.5.0
 	github.com/minio/sha256-simd v1.0.0
 	github.com/minio/simdjson-go v0.4.2

--- a/internal/config/identity/ldap/config.go
+++ b/internal/config/identity/ldap/config.go
@@ -62,6 +62,7 @@ func (l *Config) Clone() Config {
 // LDAP keys and envs.
 const (
 	ServerAddr         = "server_addr"
+	SRVRecordName      = "srv_record_name"
 	LookupBindDN       = "lookup_bind_dn"
 	LookupBindPassword = "lookup_bind_password"
 	UserDNSearchBaseDN = "user_dn_search_base_dn"
@@ -73,6 +74,7 @@ const (
 	ServerStartTLS     = "server_starttls"
 
 	EnvServerAddr         = "MINIO_IDENTITY_LDAP_SERVER_ADDR"
+	EnvSRVRecordName      = "MINIO_IDENTITY_LDAP_SRV_RECORD_NAME"
 	EnvTLSSkipVerify      = "MINIO_IDENTITY_LDAP_TLS_SKIP_VERIFY"
 	EnvServerInsecure     = "MINIO_IDENTITY_LDAP_SERVER_INSECURE"
 	EnvServerStartTLS     = "MINIO_IDENTITY_LDAP_SERVER_STARTTLS"
@@ -98,6 +100,10 @@ var (
 	DefaultKVS = config.KVS{
 		config.KV{
 			Key:   ServerAddr,
+			Value: "",
+		},
+		config.KV{
+			Key:   SRVRecordName,
 			Value: "",
 		},
 		config.KV{
@@ -173,9 +179,10 @@ func Lookup(s config.Config, rootCAs *x509.CertPool) (l Config, err error) {
 		return l, nil
 	}
 	l.LDAP = ldap.Config{
-		Enabled:    true,
-		RootCAs:    rootCAs,
-		ServerAddr: ldapServer,
+		Enabled:       true,
+		RootCAs:       rootCAs,
+		ServerAddr:    ldapServer,
+		SRVRecordName: getCfgVal(SRVRecordName),
 	}
 	l.stsExpiryDuration = defaultLDAPExpiry
 

--- a/internal/config/identity/ldap/help.go
+++ b/internal/config/identity/ldap/help.go
@@ -28,9 +28,16 @@ var (
 	Help = config.HelpKVS{
 		config.HelpKV{
 			Key:         ServerAddr,
-			Description: `AD/LDAP server address e.g. "myldapserver.com:636"` + defaultHelpPostfix(ServerAddr),
+			Description: `AD/LDAP server address e.g. "myldap.com" or "myldapserver.com:636"` + defaultHelpPostfix(ServerAddr),
 			Type:        "address",
 			Sensitive:   true,
+		},
+		config.HelpKV{
+			Key:         SRVRecordName,
+			Description: `DNS SRV record name for LDAP service, if given, must be one of "ldap", "ldaps" or "on"` + defaultHelpPostfix(SRVRecordName),
+			Optional:    true,
+			Type:        "string",
+			Sensitive:   false,
 		},
 		config.HelpKV{
 			Key:         LookupBindDN,

